packages/core/src/tools/utils.ts

```diff
@@ -146,61 +146,51 @@ export function round(num: number, decimals: 0 | 1 | 2 | 3 | 4) {
 // eslint-disable-next-line @typescript-eslint/no-empty-function
 export function noop() {}
 
-interface ObjectWithToJSON {
+interface ObjectWithToJsonMethod {
   toJSON: (() => object) | undefined
 }
+function removeToJsonMethod(object: ObjectWithToJsonMethod) {
+  const objectToJson = object.toJSON
+  if (objectToJson) {
+    delete object.toJSON
+    return () => {
+      object.toJSON = objectToJson
+    }
+  }
 
-type OriginalToJSON = [boolean, undefined | (() => object)]
+  return noop
+}
 
 /**
  * Custom implementation of JSON.stringify that ignores value.toJSON.
  * We need to do that because some sites badly override toJSON on certain objects.
- * Note this still supposes that JSON.stringify is correct...
+ * Note this still assumes that JSON.stringify is correct...
  */
 export function jsonStringify(
   value: unknown,
   replacer?: Array<string | number>,
   space?: string | number
 ): string | undefined {
-  if (value === null || value === undefined) {
-    return JSON.stringify(value)
-  }
-  let originalToJSON: OriginalToJSON = [false, undefined]
-  if (hasToJSON(value)) {
-    // We need to add a flag and not rely on the truthiness of value.toJSON
-    // because it can be set but undefined and that's actually significant.
-    originalToJSON = [true, value.toJSON]
-    delete value.toJSON
-  }
-
-  let originalProtoToJSON: OriginalToJSON = [false, undefined]
-  let prototype
-  if (typeof value === 'object') {
-    prototype = Object.getPrototypeOf(value) as object
-    if (hasToJSON(prototype)) {
-      originalProtoToJSON = [true, prototype.toJSON]
-      delete prototype.toJSON
-    }
-  }
+  // Some sites are badly overriding toJSON on certain objects. We expect that the Browser SDK won't
+  // use other objects than Array and Object to represent a value to stringify, and those objects
+  // won't have an own "toJSON" method defined.
+  //
+  // Thus, we only need to make sure that Array and Object prototypes don't define their own toJSON
+  // methods. User-provided values might have `toJSON` methods, but it is safe to keep them, since
+  // it won't impact the SDK operations or data model representation.
```

Comment:
> If a customer is giving us an object owning a toJSON method throwing an error, the current implementation works fine but with this changes it will throw

The current behavior will not change in this case. It didn't work with the current implementation because it is removing `value.toJSON` only on the *root* value, which is never a user-provided value. For example, if a user does something like this with the current version, it won't work with both current and new implementations:

```ts
DD_RUM.addRumGlobalContext('toto', { toJSON() { throw new Error('toto') } } )
```

> IIRC, the current jsonStringify implementation is inherited from the synthetics team, it could worth checking with them if they recall why this case was introduced

Ok, I'll check that

Reply:
Ah my mistake, `jsonStringify` is also used to serialize errors and console arguments, which are user-provided. I'll remove the `toJSON` method on the value then.